proxy/src/console/mgmt.rs

```diff
@@ -88,20 +90,22 @@ async fn try_process_query(pgb: &mut PostgresBackendTCP, query: &str) -> Result<
     let resp: KickSession = serde_json::from_str(query).context("Failed to parse query as json")?;
 
     let span = info_span!("event", session_id = resp.session_id);
-    let _enter = span.enter();
-    info!("got response: {:?}", resp.result);
-
-    match notify(resp.session_id, Ok(resp.result)) {
-        Ok(()) => {
-            pgb.write_message_noflush(&SINGLE_COL_ROWDESC)?
-                .write_message_noflush(&BeMessage::DataRow(&[Some(b"ok")]))?
-                .write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
-        }
-        Err(e) => {
-            error!("failed to deliver response to per-client task");
-            pgb.write_message_noflush(&BeMessage::ErrorResponse(&e.to_string(), None))?;
+    async move {
+        info!("got response: {:?}", resp.result);
+
+        match notify(resp.session_id, Ok(resp.result)) {
+            Ok(()) => {
+                pgb.write_message_noflush(&SINGLE_COL_ROWDESC)?
+                    .write_message_noflush(&BeMessage::DataRow(&[Some(b"ok")]))?
+                    .write_message_noflush(&BeMessage::CommandComplete(b"SELECT 1"))?;
+            }
+            Err(e) => {
+                error!("failed to deliver response to per-client task");
+                pgb.write_message_noflush(&BeMessage::ErrorResponse(&e.to_string(), None))?;
```

Comment:
This handling seems strange, if we failed to write something then how would we succeed in writing something more.